bluesky/run_engine.py

```diff
@@ -929,6 +949,16 @@ def _resume_task(self, *, init_func=None):
                     if (exc is not None
                             and not isinstance(exc, _RunEnginePanic)):
                         raise exc
+                    # Only try to get a result if there wasn't an error,
+                    # (other than a cancelled error)
+                    if exc is None:
```

Comment:
I don't think we need this conditional as the `if exc is not None` above will bail out of the function in all cases that would not go through this branch?

Reply:
A non-none exec could, at this point, still be a _RunEnginePanic, which would not get raised by
` if (exec is not None and not isinstance(exec, _RunEnginePanic)):`
If exec is a runtime panic, bad things happen when trying to get `self._task_fut.result()`

There may be more elegant combinations of conditionals and try/except statements, but this one at least passes all the tests, and leaving out that conditional causes it to fail on the stopped/aborted tests.